packages/replay/src/eventBuffer/EventBufferCompressionWorker.ts

```diff
@@ -0,0 +1,129 @@
+import type { ReplayRecordingData } from '@sentry/types';
+import { logger } from '@sentry/utils';
+
+import type { RecordingEvent, WorkerRequest, WorkerResponse } from '../types';
+import { EventBufferArray } from './EventBufferArray';
+
+/**
+ * Event buffer that uses a web worker to compress events.
+ * Exported only for testing.
+ */
+export class EventBufferCompressionWorker extends EventBufferArray {
+  private _worker: Worker;
+
+  private _id: number = 0;
+
+  public constructor(worker: Worker) {
+    super();
+    this._worker = worker;
+  }
+  /**
+   * Ensure the worker is ready (or not).
+   * This will either resolve when the worker is ready, or reject if an error occured.
+   */
+  public ensureReady(): Promise<void> {
+    return new Promise((resolve, reject) => {
+      this._worker.addEventListener(
+        'message',
+        ({ data }: MessageEvent) => {
+          if ((data as WorkerResponse).success) {
+            resolve();
+          } else {
+            reject();
+          }
+        },
+        { once: true },
+      );
+
+      this._worker.addEventListener(
+        'error',
+        error => {
+          reject(error);
+        },
+        { once: true },
+      );
+    });
+  }
+
+  /** @inheritdoc */
+  public destroy(): void {
+    __DEBUG_BUILD__ && logger.log('[Replay] Destroying compression worker');
+    this._worker.terminate();
+    super.destroy();
+  }
+
+  /**
+   * Finish the event buffer and return the compressed data.
+   */
+  public async finish(): Promise<ReplayRecordingData> {
+    const pendingEvents = this.pendingEvents.slice();
+
+    this.clear();
+
+    try {
+      return await this._compressEvents(this._getAndIncrementId(), pendingEvents);
```

Comment:
This change means that if something goes wrong when processing, we don't error out but just send the events uncompressed.